### PR TITLE
Contribution for #207 : uses a simple "id" instead of "name"/"BlElementId", but preserve BlElementId behavior behind to minimize impact

### PR DESCRIPTION
**Pull request by labordep**
+ add  hasChildren + add id constructor + adding syntaxic sugar + tests

https://github.com/pharo-graphics/Bloc/issues/207

### CODE REVIEW

**Comment by labordep** [comment]
Some tests are not passing, I will fix.

**Comment by labordep** [comment]
@tinchodias there is a problem with one test on Windows : 

image

This problem looks like similar to https://github.com/pharo-graphics/Bloc/issues/282

File: src/Bloc/BlElement.class.st [hunk 1]
```diff
@@ -1854,7 +1896,7 @@ BlElement >> hasChildren [
 	false otherwise"
 	<return: #Boolean>
 	
-	^ self children isEmpty not
+	^ self childrenCount > 0
```
Comment (tinchodias) [inline src/Bloc/BlElement.class.st:1899]: This is the only small change where I have a question. Is this based on some profiling or bench? to me it sounds faster asking `isEmpty` than counting

Comment (labordep) [reply src/Bloc/BlElement.class.st:1899]: Thanks @tinchodias, you have right, at the beginning I rewrote the method to remove the not necessary negation, but this is better with notEmpty

**Comment by labordep** [comment]
@tinchodias, @plantec  I'm asking me about "xxxWithId" or just "xxxId" method, what do you think is the better ?

File: src/Bloc/BlElement.class.st (hunk 1 above)
Comment (tinchodias) [reply src/Bloc/BlElement.class.st]: Okay, great. I didn't check but sometimes this kind of methods are executed very often, so a small change impacts.

**Comment by tinchodias** [comment]
If we are talking for example of the query "childWithId:", I think it would be bad idea to rename it to "childId:" because that's typically a setter/mutator. The behavior of "childWithId:" is close to "detect:" or "at:", right? if true, I think names like "childAtId:" or "detectChildWithId:" can be alternatives, but IMO "childWithId:" is fine.

**Comment by labordep** [comment]
> If we are talking for example of the query "childWithId:", I think it would be bad idea to rename it to "childId:" because that's typically a setter/mutator. The behavior of "childWithId:" is close to "detect:" or "at:", right? if true, I think names like "childAtId:" or "detectChildWithId:" can be alternatives, but IMO "childWithId:" is fine.

Thanks Martin, this is good for me.

File: src/Bloc/BlElement.class.st [hunk 2]
```diff
@@ -747,26 +753,34 @@ BlElement >> childIndexOf: anElement [
...
-		childNamed: aContainerName
+		childWithId: aContainerName
 		ifFound: ifFoundBlock
-		ifNone: [  ]
 ]
 
 { #category : #'children accessing' }
 BlElement >> childNamed: aContainerName ifFound: ifFoundBlock ifNone: ifNoneBlock [
-	<return: #Object>
+	<return: #BlElement>
```
Comment (tinchodias) [inline src/Bloc/BlElement.class.st:779]: We are gradually removing this pragma. But it's ok to keep it in this PR.

Comment (labordep) [reply src/Bloc/BlElement.class.st:779]: Ok ! I forgot to ask you about this pragma, so when I see this pragma I can remove ?

Comment (tinchodias) [reply src/Bloc/BlElement.class.st:779]: Yes. However, I'm keeping them when it's not an obvious information (like if it a testing selector is*, then it's obvious it restuns a Boolean). For a BlElement, IMO, a user should assume `childWithId:` returns a BlElement.

**Pull request merged**